src/Wordpress/CustomPostType/Item.php

```diff
@@ -293,6 +293,6 @@ public function registerMetabox() {
 		foreach ($cmb->meta_box['fields'] as $metabox_field) {
 			$metabox_fields[$metabox_field['id']] = $metabox_field['name'];
 		}
-		Settings::updateOption('commonsbooking_settings_metaboxfields', $this->getPostType(), $metabox_fields);
+		Settings::updateOption('commonsbooking_settings_metaboxfields', static::getPostType(), $metabox_fields);
```

Comment:
Das wird nicht aus einem statischen Kontext aufgerufen, warum dann nicht this verwenden?

Reply:
So wie ich [late static binding](https://www.php.net/manual/en/language.oop5.late-static-bindings.php) in php verstehe, wäre das nur relevant wenn es abgeleitete Klassen von Item gibt, welchen wir erlauben wollen `registerMetabox` mit ihrer eigenen `getPostType` Implementierung aufzurufen. Das ist hier aber nicht der Fall, ggf. war das ein false positive von phpstan für @janschoenherr

Reply:
Also bestimmt hätten wir das eindeutiger definieren können, aber mit $this wird garantiert, dass die Methode getPostType z.b. auch mit einer nicht-statischen Methode in der aktuellen Klasse überschrieben werden kann. Zumindest habe ich das so verstanden. Wir brauchen das vielleicht nicht unbedingt, das verhindert aber vielleicht in Zukunft schwer nachvollziehbare Fehler.

Reply:
Bei mir ist das eine Warnung in phpstorm. Ich finde es persönlich eher verwirrend, wenn statische Methoden nicht statisch aufgerufen werden. Als Alternative könnte man auch `$this::getPostType()` schreiben.

Ich kann das aber auch gern wieder zurück ändern.

Reply:
@hansmorb das von dir beschriebene Verhalten funktioniert auch mit `static::`. Die bisherigen Aufrufe der statischen Methode `CustomPostType::getPostType` geschehen auch auf statische Art und Weise.

Daher wäre ich doch auch für diese Änderung, hin zu dem `static`-Aufruf.